### PR TITLE
manifest: openthread: Regular openthread upmerge to `6d55738`

### PR DESCRIPTION
**Pull request by canisLupus1313**
Openthread upmerge to `6d55738`. Includes implementing of:
- `OPENTHREAD_CONFIG_RADIO_STATS_ENABLE`
- `OPENTHREAD_CONFIG_BORDER_ROUTING_DHCP6_PD_ENABLE`

### CODE REVIEW

File: modules/openthread/Kconfig.features [hunk 1]
```diff
@@ -51,6 +51,9 @@ config OPENTHREAD_BORDER_ROUTING
 config OPENTHREAD_BORDER_ROUTING_COUNTERS
 	bool "Border routing counters support"
 
+config OPENTHREAD_BORDER_ROUTING_DHCP6_PD_ENABLE
+	bool "Dhcpv6 pd support in border routing"
```
Comment (pdgendt) [inline modules/openthread/Kconfig.features:55]: Nit:
```suggestion
	bool "DHCPv6 pd support in border routing"
```

File: modules/openthread/platform/openthread-core-zephyr-config.h [hunk 2]
```diff
@@ -446,4 +446,15 @@
...
+
+ /**
+ * @def OPENTHREAD_CONFIG_RADIO_STATS_ENABLE
+ *
+ * Enable support for Radio Statistics.
+ *
+ */
+#ifndef CONFIG_OPENTHREAD_RADIO_STATS_ENABLE
+#define OPENTHREAD_CONFIG_RADIO_STATS_ENABLE CONFIG_OPENTHREAD_CONFIG_RADIO_STATS_ENABLE
+#endif
```
Comment (pdgendt) [inline modules/openthread/platform/openthread-core-zephyr-config.h:458]: Inverse logic and copy-paste issue?

```suggestion
#ifdef CONFIG_OPENTHREAD_RADIO_STATS_ENABLE
#define OPENTHREAD_CONFIG_RADIO_STATS_ENABLE CONFIG_OPENTHREAD_RADIO_STATS_ENABLE
#endif
```

**Review by pdgendt — CHANGES_REQUESTED** [review]

**Review by pdgendt — COMMENTED** [review]
Some compliance checks still need to be addressed.

File: modules/openthread/Kconfig.features [hunk 3]
```diff
@@ -51,6 +51,9 @@ config OPENTHREAD_BORDER_ROUTING
 config OPENTHREAD_BORDER_ROUTING_COUNTERS
 	bool "Border routing counters support"
 
+config OPENTHREAD_BORDER_ROUTING_DHCP6_PD_ENABLE
+	bool "DHCPv6 pd support in border routing"
```
Comment (edmont) [inline modules/openthread/Kconfig.features:55]:
```suggestion
	bool "DHCPv6-PD support in border routing"
```

**Review by edmont — APPROVED** [review]

File: modules/openthread/platform/openthread-core-zephyr-config.h [hunk 4]
```diff
@@ -446,4 +446,15 @@
 #define OPENTHREAD_CONFIG_PLATFORM_POWER_CALIBRATION_ENABLE 0
 #endif
 
+
+ /**
+ * @def OPENTHREAD_CONFIG_RADIO_STATS
```
Comment (pdgendt) [inline modules/openthread/platform/openthread-core-zephyr-config.h:451]: Nit:
```suggestion
/**
 * @def OPENTHREAD_CONFIG_RADIO_STATS
```

**Review by pdgendt — DISMISSED** [review]

**Comment by rlubos** [comment]
Seems that Babble sim tests have some trouble running. The devices start to communicate eventually, but there is some disturbance early in the test, due to both devices forming own mesh early (i. e. starting as `Leader`). One of them degrades to a `Child` after a while and joins the other mesh.

I'm not really sure why would it start to manifest on this PR though. @aescolar I'm curious, would it be possible to delay the execution of the second device for a few seconds, so that the first one has a chance to form a mesh before the other starts? Ideally, we could start the server device, wait a few seconds, and then start the client, which would then join the already existing mesh, instead of forming its own.

**Comment by aescolar** [comment]
> ... @aescolar I'm curious, would it be possible to delay the execution of the second device for a few seconds, so that the first one has a chance to form a mesh before the other starts? Ideally, we could start the server device, wait a few seconds, and then start the client, which would then join the already existing mesh, instead of forming its own.

Yes, certainly, just add  `-start_offset=2e6`  to the second device command line to delay it by 2 seconds (2 million µs) of simulated time

**Review by pdgendt — DISMISSED** [review]

File: tests/bsim/net/sockets/echo_test/tests_scripts/echo_test_802154.sh [hunk 5]
```diff
@@ -18,7 +18,7 @@ EXECUTE_TIMEOUT=100
 cd ${BSIM_OUT_PATH}/bin
 
 Execute ./bs_${BOARD}_tests_bsim_net_sockets_echo_test_prj_conf_overlay-802154_conf \
-  -v=${verbosity_level} -s=${simulation_id} -d=0 -RealEncryption=1 \
+  -v=${verbosity_level} -s=${simulation_id} -start_offset=2e6 -d=0 -RealEncryption=1 \
```
Comment (pdgendt) [inline tests/bsim/net/sockets/echo_test/tests_scripts/echo_test_802154.sh:21]: Create a separate commit for this change, please.

**Review by pdgendt — DISMISSED** [review]

**Comment by rlubos** [comment]
@aescolar Any clue why are the Babblesim tests for OT failing now? The packet exchange rate seems decent now, but the test fails with:
> d_00: @00:00:23.998575  main: The TESTCASE FAILED with return code 1

**Comment by aescolar** [comment]
Due to the 2s delay, the testcase just did not have time to pass.
https://github.com/zephyrproject-rtos/zephyr/blob/main/tests/bsim/net/sockets/echo_test/src/echo_test.c#L18
(There was only 1 second margin before).
You can just increase the maximum simulation duration by 5 seconds:

```diff
diff --git a/tests/bsim/net/sockets/echo_test/tests_scripts/echo_test_ot.sh b/tests/bsim/net/sockets/echo_test/tests_scripts/echo_test_ot.sh
index 3516426122..a5b1f856d4 100755
--- a/tests/bsim/net/sockets/echo_test/tests_scripts/echo_test_ot.sh
+++ b/tests/bsim/net/sockets/echo_test/tests_scripts/echo_test_ot.sh
@@ -25,6 +25,6 @@ Execute ./bs_${BOARD}_samples_net_sockets_echo_server_prj_conf_overlay-ot_conf\
   -v=${verbosity_level} -s=${simulation_id} -d=1 -RealEncryption=1 \
 
 Execute ./bs_2G4_phy_v1 -v=${verbosity_level} -s=${simulation_id} \
-  -D=2 -sim_length=26e6 -argschannel -at=40 -argsmain $@
+  -D=2 -sim_length=30e6 -argschannel -at=40 -argsmain $@
 
 wait_for_background_jobs #Wait for all programs in background and return != 0 if any fails
```

**Review by pdgendt — DISMISSED** [review]

**Comment by rlubos** [comment]
Nice, it seems Babblesim tests are finally passing. Thanks @canisLupus1313 for addressing the problem and thanks @aescolar for your support.

File: west.yml [hunk 6]
```diff
@@ -290,7 +290,7 @@ manifest:
       revision: c904a01d4a882bcbb39987e0e2ce5308f49ac7ad
       path: modules/lib/open-amp
     - name: openthread
-      revision: d9abe3071c0131a4adb5d7e7451319b735e6d855
+      revision: pull/97/head
```
Comment (rlubos) [inline west.yml:293]: `6d557383e8bc8654f813674e8631c2e56b8e70d0`

Comment (canisLupus1313) [reply west.yml]: done

**Review by pdgendt — APPROVED** [review]

**Review by rlubos — APPROVED** [review]

**Pull request merged**